tests/frontend/test_utils.py

```diff
@@ -341,87 +333,92 @@ def modes(self):
         return 3
 
     @pytest.mark.parametrize("pure_state", [True, False])
-    def test_random_covariance_square(self, modes, hbar, pure_state):
+    @pytest.mark.parametrize("block_diag", [True, False])
+    def test_random_covariance_square(self, modes, hbar, pure_state, block_diag):
         """Test that a random covariance matrix is the right shape"""
-        V = utils.random_covariance(modes, hbar=hbar, pure=pure_state)
+        V = utils.random_covariance(modes, hbar=hbar, pure=pure_state, block_diag=block_diag)
         assert np.all(V.shape == np.array([2 * modes, 2 * modes]))
```

Comment:
nice tests!

Reply:
Nice, `pytest` does all the work here :)